### PR TITLE
User and followers

### PR DESCRIPTION
**Pull request by anowakowski**

### CODE REVIEW

File: PictureApp.API/Controllers/FollowersController.cs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        public FollowersController(IUserService userService, IFollowerService followerService)
+        {
+            this.userService = userService;
+            _followerService = followerService;
+        }
+
+        [HttpPost("{id}/setfollow")]
+        public async Task<IActionResult> SetUpFollower(int userId, int id)
+        {
+            if (userId != int.Parse(User.FindFirst(ClaimTypes.NameIdentifier).Value))
```
Comment (mvein) [inline PictureApp.API/Controllers/FollowersController.cs]: In this case we do authorization check twice. First time here and the second one in _followerService.SetUpFollower.

File: PictureApp.API.Services/FollowerService.cs [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+            }
+        }
+
+        private async Task ValidateUser(int userId, int recipientId)
+        {
+            var user = await _userService.GetUser(userId);
+            var recpitent = await _userService.GetUser(recipientId);
+
+            if (user == null)
+                throw new NotAuthorizedException($"user by {userId} not found");
```
Comment (mvein) [inline PictureApp.API.Services/FollowerService.cs]: Why for lack of user we use different exception?

File: PictureApp.API.Services/FollowerService.cs [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+            _repository = repository;
+            _userRepository = userRepository;
+            _unitOfWork = unitOfWork;
+            _mapper = mapper;
+        }
+
+        public async Task<IEnumerable<UsersListWithFollowersForExploreDto>> GetAllWithFollowers(int userId)
+        {   
+            var usersFollower = await _repository.FindAsync(x => x.FollowerId == userId);
+            var users = await _userRepository.GetAllAsync();
```
[the remark this answers is not in the document]
Comment (anowakowski) [reply PictureApp.API.Services/FollowerService.cs]: Yes, we talk about this several day ago. Its temporary solution. We should think about how to extract user correctly. But for this follower functionality and this case is suitable approach.

File: PictureApp.API/Controllers/FollowersController.cs [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+using PictureApp.API.Services;
+
+namespace PictureApp.API.Controllers
+{
+    [Authorize]
+    [Route("api/users/{userId}/followers")]
+    [ApiController]
+    public class FollowersController : ControllerBase
+    {
+        private readonly IUserService userService;
```
[the remark this answers is not in the document]
Comment (anowakowski) [reply PictureApp.API/Controllers/FollowersController.cs]: ok good point. I will fixed name convention

File: PictureApp.API/Controllers/FollowersController.cs [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        private readonly IFollowerService _followerService;
+
+        public FollowersController(IUserService userService, IFollowerService followerService)
+        {
+            this.userService = userService;
```
[the remark this answers is not in the document]
Comment (anowakowski) [reply PictureApp.API/Controllers/FollowersController.cs]: ok good point. I will fixed name convention

File: PictureApp.API.Services/FollowerService.cs (hunk 2 above)
Comment (anowakowski) [reply PictureApp.API.Services/FollowerService.cs]: fixed by:
https://github.com/anowakowski/PictureApp_API/pull/29/commits/75558ef8a39ce9d95f1a826d45e2793b5e1a8440

File: PictureApp.API/Controllers/FollowersController.cs (hunk 1 above)
Comment (anowakowski) [reply PictureApp.API/Controllers/FollowersController.cs]: fixed by:
https://github.com/anowakowski/PictureApp_API/pull/29/commits/75558ef8a39ce9d95f1a826d45e2793b5e1a8440

File: PictureApp.API/Controllers/FollowersController.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (anowakowski) [reply PictureApp.API/Controllers/FollowersController.cs]: fixed by:
https://github.com/anowakowski/PictureApp_API/pull/29/commits/75558ef8a39ce9d95f1a826d45e2793b5e1a8440

File: PictureApp.API/Controllers/FollowersController.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (anowakowski) [reply PictureApp.API/Controllers/FollowersController.cs]: fixed by:
https://github.com/anowakowski/PictureApp_API/pull/29/commits/75558ef8a39ce9d95f1a826d45e2793b5e1a8440

File: PictureApp.API.Services/FollowerService.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (mvein) [reply PictureApp.API.Services/FollowerService.cs]: Sorry, I had to missed/forgot this talk.

File: PictureApp.API.Services/FollowerService.cs [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+            }
+            else
+            {
+                ////log if user arleady unfollow choosen recipient
+            }
+        }
+
+        private async Task ValidateUser(int userId, int recipientId)
+        {
+            var user = await _userService.GetUser(userId);
```
[the remark this answers is not in the document]
Comment (mvein) [reply PictureApp.API.Services/FollowerService.cs]: Ok, let's leave this for future refactoring ;)

**Comment by mvein** [comment]
My comments above.

**Pull request closed without merging**